src/mixins/auth-api-mixin.ts

```diff
@@ -16,23 +15,95 @@ export default class AuthApiMixin extends Vue {
   @Getter getBusinessId!: string
 
   /**
-   * Update a businesses contact information.
-   * @param contactInfo The contact information payload.
+   * Fetches authorizations.
+   * @param businessIdentifier the business identifier (eg, BC1219948)
+   * @returns a promise to return the authorizations object
+   */
+  async fetchAuthorizations (businessIdentifier: string): Promise<any> {
+    if (!businessIdentifier) throw new Error('Invalid parameter \'businessIdentifier\'')
+
+    const url = `entities/${businessIdentifier}/authorizations`
+    const authUrl = sessionStorage.getItem('AUTH_API_URL')
+    const config = { baseURL: authUrl }
+
+    return axios.get(url, config)
+  }
+
+  /**
+   * Fetches the current user data.
+   * @returns a promise to return the data
+   */
+  async fetchCurrentUser (): Promise<any> {
+    const authUrl = sessionStorage.getItem('AUTH_API_URL')
+    const config = { baseURL: authUrl }
+
+    return axios.get('users/@me', config)
+  }
+
+  /**
+   * Fetches the auth info of the current business.
+   * @returns a promise to return the data
+   */
+  async fetchAuthInfo (): Promise<AuthInformationIF> {
+    if (!this.getBusinessId) throw new Error('Invalid business id')
+
+    const url = `entities/${this.getBusinessId}`
+    const authUrl = sessionStorage.getItem('AUTH_API_URL')
+    const config = { baseURL: authUrl }
+
+    return axios.get(url, config)
+      .then(response => {
+        if (response?.data) {
+          return {
+            contacts: response.data.contacts,
+            folioNumber: response.data.folioNumber
+          }
+        } else {
+          // eslint-disable-next-line no-console
+          console.log('fetchAuthInfo() error - invalid response =', response)
+          throw new Error('Invalid API response')
+        }
+      })
+  }
+
+  /**
+   * Updates the businesses contact information.
+   * @param contactInfo the contact information object
    */
   updateContactInfo (contactInfo: ContactPointIF): Promise<any> {
-    try {
-      const contactPayload = {
-        email: contactInfo.email,
-        phone: contactInfo.phone,
-        phoneExtension: contactInfo.extension
-      }
-      const authUrl = sessionStorage.getItem('AUTH_API_URL')
-      let url = `${authUrl}entities/${this.getBusinessId}/contacts`
-
-      return axios.put(url, contactPayload)
-    } catch (error) {
-      // TODO: Update error handling modal after design review
-      this.$root.$emit('save-error-event', error)
+    if (!this.getBusinessId) throw new Error('Invalid business id')
+
+    const data = {
+      email: contactInfo.email,
+      phone: contactInfo.phone,
+      phoneExtension: contactInfo.extension
     }
+    const authUrl = sessionStorage.getItem('AUTH_API_URL')
+    let url = `${authUrl}entities2/${this.getBusinessId}/contacts`
+
+    return axios.put(url, data)
+      .catch(error => {
```

Comment:
This was previously wrapped in a try-catch block, which did not catch errors. Fixed.